tests/test_properties.py

```diff
@@ -0,0 +1,416 @@
+
+    assert np.linalg.det(np.cov(X)) > np.linalg.det(np.cov(params_es))
+
+    params_ies = X.copy()
+    responses_ies = response_ensemble.copy()
+    smoother_ies = ies.SIES(ensemble_size)
+    for _ in range(10):
+        smoother_ies.fit(responses_ies, d.sd.values, d.value.values, noise=E)
+        params_ies = smoother_ies.update(params_ies)
+
+        _coeff_a = params_ies[0, :]
+        _coeff_b = params_ies[1, :]
+        _coeff_c = params_ies[2, :]
+
+        _fwd_runs = p_map(
+            poly,
+            _coeff_a,
+            _coeff_b,
+            _coeff_c,
+            [np.arange(max(x_observations) + 1)] * ensemble_size,
+            desc="SIES ert iteration",
+        )
+
+        assert np.linalg.det(np.cov(X)) > np.linalg.det(np.cov(params_ies))
+
+        responses_ies = np.array(
+            [fwd_run[d.index.get_level_values("x").to_list()] for fwd_run in _fwd_runs]
+        ).T
+
+    assert np.abs(
+        (np.linalg.det(np.cov(params_es)) - np.linalg.det(np.cov(params_ies))) < 0.001
+    )
+
+
+var = 2.0
+
+
+@pytest.mark.parametrize(
+    "inversion,errors",
+    [
+        (ies.InversionType.EXACT, np.diag(np.array([var, var, var]))),
+        (ies.InversionType.EXACT, np.array([np.sqrt(var), np.sqrt(var), np.sqrt(var)])),
+        (ies.InversionType.EXACT_R, np.diag(np.array([var, var, var]))),
+        (
+            ies.InversionType.EXACT_R,
+            np.array([np.sqrt(var), np.sqrt(var), np.sqrt(var)]),
+        ),
+        (ies.InversionType.SUBSPACE_RE, np.diag(np.array([var, var, var]))),
+        (
+            ies.InversionType.SUBSPACE_RE,
+            np.array([np.sqrt(var), np.sqrt(var), np.sqrt(var)]),
+        ),
+    ],
+)
+def test_that_update_correctly_multiples_gaussians(inversion, errors):
+    """
+    NB! This test is potentially flaky because of finite ensemble size.
+
+    Bayes' theorem states that
+    p(x|y) is proportional to p(y|x)p(x)
+
+    Assume p(x) is N(mu=0, Sigma=2I) and p(y|x) is N(mu=y, Sigma=2I).
+    Multiplying these together (see 8.1.8 of the matrix cookbook) we get
+    that p(x|y) is N(mu=y/2, Sigma=I).
+    Note that Sigma is a covariance matrix.
+
+    Here we use this property, and assume that the forward model is the identity
+    to test analysis steps.
+    """
+    N = 1500
+    nparam = 3
+
+    A = rng.standard_normal(size=(nparam, N))
+    A = np.linalg.cholesky(var * np.identity(nparam)) @ A
+    # Assuming forward model is the identity
+    Y = A
+
+    obs_val = 10
+    observation_values = np.array([obs_val, obs_val, obs_val])
+    smoother = ies.SIES(N)
+    smoother.fit(
+        Y,
+        errors,
+        observation_values,
+        inversion=inversion,
+        step_length=1.0,
+        truncation=1.0,
+    )
+    A_ES = smoother.update(A)
+
+    for i in range(nparam):
+        assert np.isclose(A_ES[i, :].mean(), obs_val / 2, rtol=0.15)
+
+    assert (np.abs(np.cov(A_ES) - np.identity(nparam)) < 0.15).all()
+
+
+def test_make_D():
```

Comment:
Is this Equation 9.22 or Equation 9.24 in Evensen 2009? Could we add a comment?
From reading [the implementation](https://github.com/dafeda/iterative_ensemble_smoother/blob/9ec40e6f4b622f75d118cca637532bdee8868023/src/iterative_ensemble_smoother/ies.cpp#L292) it seems to be 9.24.

![image](https://user-images.githubusercontent.com/13650516/221889545-f3244132-d240-498f-94a3-413ffe873438.png)

Reply:
It is equation 9.24 yes.
We could add a comment, but perhaps it's better to create an issue where we use `H` instead of `Dprime` as `H` is used in the paper.
What do you think?

Reply:
I think creating an issue is the better solution, given that the notation of the 2019 paper should be used (seems natural) over the 2009 book. 

I took the liberty of creating the issue and will also be happy to follow it up https://github.com/equinor/iterative_ensemble_smoother/issues/75

Reply:
That's great 👍